pkg/networkservice/common/connect/networkservice_client.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connect
+
+import (
+	"context"
+	"net/url"
+
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/clienturl"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/heal"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/adapters"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/chain"
+)
+
+// NewNetworkServiceClient - adapts connect server chain to client
+func NewNetworkServiceClient(
+	ctx context.Context,
+	connectTo *url.URL,
+	clientFactory ClientFactory,
+	clientOpts ...Option,
+) networkservice.NetworkServiceClient {
+	var opts = &connectServer{}
+	for _, opt := range clientOpts {
+		opt(opts)
+	}
+	return adapters.NewServerToClient(
+		chain.NewNetworkServiceServer(
+			clienturl.NewServer(connectTo),
+			heal.NewServer(ctx, nil),
```

Comment:
This is totally incorrect:
1. `connect` -> `heal` dependency
2. heal server should take a pointer to the whole chain as a 2 parameter

Reply:
```go
rv := new(networkservice.NetworkServiceClient)
	*rv = chain.NewNetworkServiceClient(
		updatepath.NewClient(opts.name),
		serialize.NewClient(),
		refresh.NewClient(ctx),
		metadata.NewClient(),
		adapters.NewServerToClient(
			chain.NewNetworkServiceServer(
				heal.NewServer(ctx, rv),
				clienturl.NewServer(connectTo),
				connect.NewServer(ctx, func(ctx context.Context, cc grpc.ClientConnInterface) networkservice.NetworkServiceClient {
					return chain.NewNetworkServiceClient(
						append(
							opts.additionalFunctionality,
							heal.NewClient(ctx, networkservice.NewMonitorConnectionClient(cc)),
							opts.authorizeClient,
							networkservice.NewNetworkServiceClient(cc),
						)...,
					)
				},
					connect.WithDialOptions(opts.dialOptions...),
					connect.WithDialTimeout(opts.dialTimeout)),
			),
		),
	)

	return *rv
```